test/ParserSpec.hs

```diff
@@ -1,19 +1,54 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+
 module ParserSpec where
 
 import           Control.Monad.Reader
 import           Data.Attoparsec.ByteString.Char8
 import qualified Data.ByteString.Char8 as BS8
 import           Data.Either
 import qualified Data.Map as M
-import           Test.Hspec
+import           Data.Aeson
+import           Test.Hspec hiding (Arg)
 import           Test.QuickCheck
-import           TestData
 import           Text.PrettyPrint.HughesPJ
 import           Weft.Generics.PprQuery
 import           Weft.Generics.QueryParser
 import           Weft.Internal.Types
 import           Weft.Types
 
+newtype Id = Id String deriving (Generic, Show, Read, Eq, Ord, Arbitrary, ToJSON)
+newtype Name = Name String deriving (Generic, Show, Eq, Ord, Arbitrary, ToJSON)
+
+data User ts = User
```

Comment:
unrelated to your PR, just thinking aloud:

Gee this is a lot of boilerplate just to model a simple schema. Big constant factors for all the `Magic` machinery, and then all of the standalone deriving crap. What do you think if we encoded `User` as this instead:

```haskell
data User = User
  { userId :: WithArgs UserIdArgs Id
  , userName :: Name
  , userBestFriend :: WithArgs UserBestFriendArgs User
  , userFriends :: [User]
  } deriving (Generic, Show, Eq, ToJSON, Arbitrary)
```

and then replace `User 'Data` with `HKD User Bare` (we could come up with a friendlier looking type, eg `RecordOf User`).

Reply:
That sounds way better